x-pack/plugins/actions/server/builtin_action_types/xmatters.ts

```diff
@@ -0,0 +1,329 @@
+import { map, getOrElse } from 'fp-ts/lib/Option';
+import { getRetryAfterIntervalFromHeaders } from './lib/http_rersponse_retry_header';
+import { isOk, Result } from './lib/result_type';
+import { ActionType, ActionTypeExecutorOptions, ActionTypeExecutorResult } from '../types';
+import { ActionsConfigurationUtilities } from '../actions_config';
+import { Logger } from '../../../../../src/core/server';
+import { postXmatters } from './lib/post_xmatters';
+
+export type XmattersActionType = ActionType<
+  ActionTypeConfigType,
+  ActionTypeSecretsType,
+  ActionParamsType,
+  unknown
+>;
+export type XmattersActionTypeExecutorOptions = ActionTypeExecutorOptions<
+  ActionTypeConfigType,
+  ActionTypeSecretsType,
+  ActionParamsType
+>;
+
+const configSchemaProps = {
+  url: schema.string(),
+  hasAuth: schema.boolean({ defaultValue: true }),
+};
+const ConfigSchema = schema.object(configSchemaProps);
+export type ActionTypeConfigType = TypeOf<typeof ConfigSchema>;
+
+// secrets definition
+export type ActionTypeSecretsType = TypeOf<typeof SecretsSchema>;
+const secretSchemaProps = {
+  user: schema.nullable(schema.string()),
+  password: schema.nullable(schema.string()),
+};
+const SecretsSchema = schema.object(secretSchemaProps, {
+  validate: (secrets) => {
+    // user and password must be set together (or not at all)
+    if (!secrets.password && !secrets.user) return;
+    if (secrets.password && secrets.user) return;
+    return i18n.translate('xpack.actions.builtin.xmatters.invalidUsernamePassword', {
+      defaultMessage: 'both user and password must be specified',
+    });
+  },
+});
+
+// params definition
+export type ActionParamsType = TypeOf<typeof ParamsSchema>;
+const ParamsSchema = schema.object({
+  alertActionGroup: schema.maybe(schema.string()),
+  alertActionGroupName: schema.string(),
+  alertId: schema.string(),
+  alertInstanceId: schema.maybe(schema.string()),
+  alertName: schema.maybe(schema.string()),
+  date: schema.maybe(schema.string()),
+  severity: schema.maybe(schema.string()),
+  spaceId: schema.maybe(schema.string()),
+  tags: schema.maybe(schema.string()),
+});
+
+export const ActionTypeId = '.xmatters';
+// action type definition
+export function getActionType({
+  logger,
+  configurationUtilities,
+}: {
+  logger: Logger;
+  configurationUtilities: ActionsConfigurationUtilities;
+}): XmattersActionType {
+  return {
+    id: ActionTypeId,
+    minimumLicenseRequired: 'gold',
+    name: i18n.translate('xpack.actions.builtin.xmattersTitle', {
+      defaultMessage: 'xMatters',
+    }),
+    validate: {
+      config: schema.object(configSchemaProps, {
+        validate: curry(validateActionTypeConfig)(configurationUtilities),
+      }),
+      secrets: SecretsSchema,
+      params: ParamsSchema,
+    },
+    renderParameterTemplates,
+    executor: curry(executor)({ logger, configurationUtilities }),
+  };
+}
+
+function renderParameterTemplates(params: ActionParamsType): ActionParamsType {
```

Comment:
This no-op function is _I think_ why the action variables are not being expanded.  Most connectors don't need this, but email, es-index, slack and webhook all have fields with special escaping concerns, and this function allows them to customize the escaping.  If the function isn't provided, you only get the "default" escaping, which I think will be good enough for xMatters.  So you can just remove this function, and the reference to it, and I think the action variables will start getting expanded

Reply:
that fixed it! Thank you